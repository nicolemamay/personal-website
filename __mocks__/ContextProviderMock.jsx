import React from 'react';
import Context from 'src/context';

const ContextProviderMock = ({ children, isIntroDone }) => {
  const setState = jest.fn();
  const state = {
    isIntroDone: isIntroDone,
  };

  return (
    <Context.Provider value={{ state, setState }}>{children}</Context.Provider>
  );
};

export default ContextProviderMock;
