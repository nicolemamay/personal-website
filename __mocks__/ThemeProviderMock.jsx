import React from 'react';
import { ThemeProvider } from 'styled-components';

import Theme from 'src/styles/Theme';

const ThemeProviderMock = ({ children }) => (
  <ThemeProvider theme={Theme}>{children}</ThemeProvider>
);

export default ThemeProviderMock;
