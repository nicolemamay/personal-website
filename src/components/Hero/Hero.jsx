import React, { useContext, useEffect } from 'react';
import { motion, useAnimation } from 'framer-motion';
import PropTypes from 'prop-types';

import Context from 'src/context';
import useHeroQuery from 'src/hooks/useHeroQuery';
import ContentWrapper from 'src/styles/ContentWrapper';

import SplashScreen from 'src/components/SplashScreen/SplashScreen';

import { StyledSubtitle } from './Hero.styled';

export const PureHero = ({ greetings, isIntroDone, subtitle, title }) => {
  // Required for animation - start after the splashScreen sequence is done
  const controls = useAnimation();

  useEffect(() => {
    if (isIntroDone) {
      controls.start({ opacity: 1, y: 0, transition: { delay: 0.4 } });
    }
  }, [isIntroDone, controls]);

  return (
    <section>
      {!isIntroDone && <SplashScreen />}
      {isIntroDone && (
        <ContentWrapper>
          <motion.div animate={controls} initial={{ opacity: 0, y: 20 }}>
            <h1>
              {greetings}
              <br />
              {title}
            </h1>
            <StyledSubtitle className="text-secondary">
              {subtitle}
            </StyledSubtitle>
          </motion.div>
        </ContentWrapper>
      )}
    </section>
  );
};

PureHero.propTypes = {
  greetings: PropTypes.string.isRequired,
  isIntroDone: PropTypes.bool.isRequired,
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

const Hero = () => {
  const { greetings, subtitle, title } = useHeroQuery();
  const { isIntroDone } = useContext(Context).state;

  return (
    <PureHero
      greetings={greetings}
      subtitle={subtitle}
      title={title}
      isIntroDone={isIntroDone}
    />
  );
};

export default Hero;
