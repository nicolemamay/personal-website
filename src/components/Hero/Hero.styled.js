import styled from 'styled-components';

export const StyledSubtitle = styled.p`
  && {
    max-width: 300px;
    @media (min-width: ${({ theme }) => theme.breakpoints.sm}) {
      max-width: none;
    }
  }
`;
