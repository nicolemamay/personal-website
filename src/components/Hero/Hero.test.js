import React from 'react';
import { render } from '@testing-library/react';

import ContextProviderMock from '__mocks__/ContextProviderMock';
import ThemeProviderMock from '__mocks__/ThemeProviderMock';

import { PureHero as Hero } from './Hero';

describe('Hero', () => {
  it('renders default', () => {
    // Created using the query from Hero.js
    const mockData = {
      greetings: 'Test greeting,',
      subtitle: 'Test subtitle',
      title: 'Test title',
      isIntroDone: true,
    };

    const { container } = render(
      <ContextProviderMock isIntroDone={false}>
        <ThemeProviderMock>
          <Hero {...mockData} />
        </ThemeProviderMock>
      </ContextProviderMock>,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('Hero', () => {
  it('renders splash screen', () => {
    // Created using the query from Hero.js
    const mockData = {
      greetings: 'Test greeting,',
      subtitle: 'Test subtitle',
      title: 'Test title',
      isIntroDone: false,
    };

    const { container } = render(
      <ContextProviderMock isIntroDone={true}>
        <ThemeProviderMock>
          <Hero {...mockData} />
        </ThemeProviderMock>
      </ContextProviderMock>,
    );

    expect(container).toMatchSnapshot();
  });
});
