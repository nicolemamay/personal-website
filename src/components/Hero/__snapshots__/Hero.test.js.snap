// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders default 1`] = `
<div>
  <section>
    <div
      class="sc-bdfBwQ dMFlpw"
    >
      <div
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        <h1>
          Test greeting,
          <br />
          Test title
        </h1>
        <p
          class="sc-jSgupP dScaXo text-secondary"
        >
          Test subtitle
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`Hero renders splash screen 1`] = `
<div>
  <section>
    <div
      class="sc-eCssSg kOKYJQ"
      style="opacity: 0;"
    >
      <div
        class="sc-hKgILt kwoswD"
      >
        <div
          class="sc-dlfnbm eDIRJk"
          style="height: 100%;"
        />
        <div
          class="sc-gsTCUz kPOeUf"
          color="white"
        >
          nm.
        </div>
      </div>
    </div>
  </section>
</div>
`;
